### PR TITLE
*: make the id allocator more readable 

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve?

make code more readable.

### Check List

Tests

- Unit test

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tidb-ansible`](https://github.com/pingcap/tidb-ansible):
- Need to cherry-pick to the release branch

### CODE REVIEW

File: server/id/id.go [hunk 1]
```diff
@@ -44,8 +49,8 @@ type AllocatorImpl struct {
 	member   string
 }
 
-// NewAllocatorImpl creates a new IDAllocator.
-func NewAllocatorImpl(client *clientv3.Client, rootPath string, member string) *AllocatorImpl {
+// NewAllocator creates a new IDAllocator.
```
Comment (JmPotato) [inline server/id/id.go:52]:
```suggestion
// NewAllocator creates a new ID Allocator.
```

File: server/id/id.go [hunk 2]
```diff
@@ -44,8 +49,8 @@ type AllocatorImpl struct {
 	member   string
 }
 
-// NewAllocatorImpl creates a new IDAllocator.
-func NewAllocatorImpl(client *clientv3.Client, rootPath string, member string) *AllocatorImpl {
+// NewAllocator creates a new ID Allocator.
+func NewAllocator(client *clientv3.Client, rootPath string, member string) Allocator {
 	return &AllocatorImpl{client: client, rootPath: rootPath, member: member}
```
Comment (HunDunDM) [inline server/id/id.go:54]: Need to become private type?

**Review by HunDunDM — COMMENTED** [review]
rest LGTM

**Comment by disksing** [comment]
Noce work! `Rebase` is a much better name than `Generate`.
I'm curious about which specific PR introduced the name `Generate`?

**Review by HunDunDM — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

File: pkg/mock/mockid/mockid.go [hunk 3]
```diff
@@ -29,3 +29,8 @@ func NewIDAllocator() *IDAllocator {
 func (alloc *IDAllocator) Alloc() (uint64, error) {
 	return atomic.AddUint64(&alloc.base, 1), nil
 }
+
+// Rebase implements the IDAllocator interface.
+func (alloc *IDAllocator) Rebase() error {
```
Comment (AndreMouche) [inline pkg/mock/mockid/mockid.go:34]: Nice 👍

**Review by AndreMouche — APPROVED** [review]

**Pull request merged**